pkg/util/utils.go

```diff
@@ -244,3 +270,110 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx *context.Context, k8s
 	}
 	return nil
 }
+
+// GenerateID generate id for nsx resource, some resources has complex index, so set it type to string
+func GenerateID(res_id, prefix, suffix string, index string) string {
+	var id strings.Builder
+	if len(prefix) > 0 {
+		id.WriteString(prefix)
+		id.WriteString("_")
+	}
+
+	id.WriteString(res_id)
+	if len(index) > 0 {
+		id.WriteString("_")
+		id.WriteString(index)
+
+	}
+	if len(suffix) > 0 {
+		id.WriteString("_")
+		id.WriteString(suffix)
+	}
+	return id.String()
+}
+
+func GenerateDisplayName(res_name, prefix, suffix, project, cluster string) string {
+	var name strings.Builder
+	if len(prefix) > 0 {
+		name.WriteString(prefix)
+		name.WriteString("-")
+	}
+	if len(project) > 0 {
+		name.WriteString(project)
+		name.WriteString("-")
+	}
+	if len(cluster) > 0 {
+		name.WriteString(cluster)
+		name.WriteString("-")
+	}
+	name.WriteString(res_name)
+	if len(suffix) > 0 {
+		name.WriteString("-")
+		name.WriteString(suffix)
+	}
+	return name.String()
+}
+
+func GenerateTruncName(limit int, res_name, prefix, suffix, project, cluster string) string {
+	name := GenerateDisplayName(res_name, prefix, suffix, project, cluster)
+	return normalizeNamebyLimit(name, limit)
```

Comment:
For now, when you generate the truncated name, the steps are generate full name with prefix/suffix/project/cluster -> limit the length, right? But that may lose some info when it is truncated, e.g. the suffix. I think the better sequence should be limit the length of every part firstly, then generate the full name, like what we did in NCP. How do you think?

Reply:
done